pages/c/_cluster/fleet/GitRepoGraphConfig.js

```diff
@@ -0,0 +1,248 @@
+     */
+  watcherProp: 'data.bundles',
+  /**
+     * Mandatory params for a child object in parseData (for statuses to work)
+     * @param {String} state
+     * @param {String} stateDisplay
+     * @param {String} stateColor
+     * @param {String} matchingId (this can be different than the actual ID, depends on the usecase)
+     */
+  parseData:   (data) => {
+    const bundles = data.bundles.map((bundle, i) => {
+      const bundleLowercaseState = bundle.state ? bundle.state.toLowerCase() : 'unknown';
+      const bundleStateColor = STATES[bundleLowercaseState].color;
+
+      const repoChild = {
+        id:             bundle.id,
+        matchingId:     bundle.id,
+        type:           bundle.type,
+        state:          bundle.state,
+        stateLabel:     bundle.stateDisplay,
+        stateColor:     bundleStateColor,
+        isBundle:       true,
+        errorMsg:       bundle.stateDescription,
+        detailLocation: bundle.detailLocation,
+        children:       []
+      };
+
+      const bds = data.bundleDeployments.filter(bd => bundle.id === `${ bd.metadata?.labels?.['fleet.cattle.io/bundle-namespace'] }/${ bd.metadata?.labels?.['fleet.cattle.io/bundle-name'] }`);
+
+      bds.forEach((bd) => {
+        const bdLowercaseState = bd.state ? bd.state.toLowerCase() : 'unknown';
+        const bdStateColor = STATES[bdLowercaseState].color;
+
+        const cluster = data.clustersList.find((cluster) => {
+          const clusterString = `${ cluster.namespace }-${ cluster.name }`;
+
+          return bd.id.includes(clusterString);
+        });
+
+        repoChild.children.push({
+          id:                    bd.id,
+          matchingId:            bd.id,
+          type:                  bd.type,
+          clusterId:             cluster ? cluster.id : undefined,
+          clusterDetailLocation: cluster ? cluster.detailLocation : undefined,
+          state:                 bd.state,
+          stateLabel:            bd.stateDisplay,
+          stateColor:            bdStateColor,
+          isBundleDeployment:    true,
+          errorMsg:              bd.stateDescription,
+          detailLocation:        bd.detailLocation,
+        });
+      });
+
+      return repoChild;
+    });
+
+    const repoLowercaseState = data.state ? data.state.toLowerCase() : 'unknown';
+    const repoStateColor = STATES[repoLowercaseState].color;
+
+    const finalData = {
+      id:             data.id,
+      matchingId:     data.id,
+      type:           data.type,
+      state:          data.state,
+      stateLabel:     data.stateDisplay,
+      stateColor:     repoStateColor,
+      isRepo:         true,
+      errorMsg:       data.stateDescription,
+      detailLocation: data.detailLocation,
+      children:       bundles
+    };
+
+    return finalData;
+  },
+  /**
+     * Used to add relevant classes to each main node instance
+     */
+  extendNodeClass: (d) => {
```

Comment:
It looks like we just need `data` from the input parameter, we could probably destructure it as `({data}) => { doSomething() }` rather than name the input parameter as `(d) => { doSomething() }` which allows us to avoid single letter variable names in this case.

Reply:
Thanks for the heads up. Will change this